frontend/src/test/scala/bloop/dap/DebugAdapterConnection.scala

```diff
@@ -0,0 +1,82 @@
+package bloop.dap
+
+import java.net.{InetSocketAddress, Socket, URI}
+
+import bloop.dap.DebugTestEndpoints._
+import com.microsoft.java.debug.core.protocol.Events
+import com.microsoft.java.debug.core.protocol.Requests._
+import com.microsoft.java.debug.core.protocol.Types.Capabilities
+import monix.eval.Task
+import monix.execution.Scheduler
+
+/**
+ * Manages a connection with a debug adapter.
+ * It closes the connection after receiving a response to the 'disconnect' request
+ */
+private[dap] final class DebugAdapterConnection(val socket: Socket)(
+    implicit proxy: DebugSessionProxy
+) {
+  def initialize(): Task[Capabilities] = {
+    val arguments = new InitializeArguments()
+    Initialize(arguments)
+  }
+
+  def configurationDone(): Task[Unit] = {
+    ConfigurationDone(())
```

Comment:
Is this instantiation sending a request and then waiting for a response? If that's the case, can we avoid these side effects and have a clearer API?

Reply:
You mean instantiation of `()` ? I don't believe I follow :(

Reply:
`ConfigurationDone` returns a `Task` and inside that task it does lots of IO. From reading the code, I would have never guessed that the instantiation of a class would trigger the creation and delivery of a request and then the blocking on a response.
  
`Request` should not be doing all of that work inside `apply`. My suggestion is that we decouple clearly the creation of a request with its inputs and the time when that request is sent. These are two different responsibilities that merit different API endpoints.